Sources/StreamChat/APIClient/APIClient_Tests.swift

```diff
@@ -282,43 +180,28 @@ class APIClient_Tests: StressTestCase {
         // Check the outgoing error from the encoder is the result data
         AssertResultFailure(result, encoderError)
     }
-
-    func test_uploadFileFailure() throws {
-        // Create a test request and set it as a response from the encoder
-        let testRequest = URLRequest(url: .unique())
-        encoder.encodeRequest = .success(testRequest)
-
-        // We cannot use `TestError` since iOS14 wraps this into another error
-        let networkError = NSError(domain: "TestNetworkError", code: -1, userInfo: nil)
-        let encoderError = TestError()
-
-        // Set up a mock network response from the request
-        MockNetworkURLProtocol.mockResponse(request: testRequest, statusCode: 444, error: networkError)
-
-        // Set up a decoder response to return `encoderError`
-        decoder.decodeRequestResponse = .failure(encoderError)
-
-        // Create a test endpoint (it's actually ignored, because APIClient uses the testRequest returned from the encoder)
-        let testEndpoint: Endpoint<FileUploadPayload> = .uploadAttachment(with: .unique, type: .image)
-
-        // Create a request and wait for the completion block
-        let result = try waitFor {
-            apiClient.uploadFile(
-                endpoint: testEndpoint,
-                multipartFormData: .init(Data(), fileName: .unique),
-                completion: $0
-            )
-        }
-
-        // Check the incoming error to the encoder is the error from the response
-        XCTAssertNotNil(decoder.decodeRequestResponse_error)
-
-        // We have to compare error codes, since iOS14 wraps network errors into `NSURLError`
-        // in which we cannot retrieve the wrapper error
-        XCTAssertEqual((decoder.decodeRequestResponse_error as NSError?)?.code, networkError.code)
-
-        // Check the outgoing data is from the decoder
-        XCTAssertEqual(result.error as? TestError, encoderError)
+    
+    func test_uploadAttachment_calls_CDNClient() throws {
+        let attachment = ChatMessageAttachment.sample()
+        
+        var progress: Double?
+        var result: Result<URL, Error>?
+        
+        apiClient.uploadAttachment(
+            attachment,
+            progress: { progress = $0 },
+            completion: { result = $0 }
+        )
+        
+        XCTAssertTrue(cdnClient.uploadAttachmentMockFunc.called)
+        
+        let parameters = try XCTUnwrap(cdnClient.uploadAttachmentMockFunc.calls.last)
+        
+        parameters.1?(42)
+        XCTAssertEqual(progress, 42)
+        
+        parameters.2(.success(.sampleYodaImage))
+        XCTAssertEqual(try result?.get(), .sampleYodaImage)
```

Comment:
`42` and `sampleYogaImage` shouldn't be hardcoded IMHO.

Reply:
Fair, extracted to variables

Reply:
I don't know how to handle progress, but `result` should be like
```swift
XCTAssertEqual(try result?.get(), attachment.file)
```
or something

Reply:
Not really I'm afraid, it has nothing to do with attachment.file
Basically we just make sure that completions we pass to APIClient it then proxies to CDNClient